src/Homely.Storage.Blobs/IBlob.cs

```diff
@@ -13,28 +13,44 @@ public interface IBlob
         /// Streams an item from Azure Blob storage.
         /// </summary>
         /// <remarks>No custom json-deserialization occurs here. Just raw retrevial via stream.</remarks>
-        /// <param name="blobName">string: The identifier/name of this content to be stored on Azure.</param>
+        /// <param name="blobId">string: The identifier/name of this content to be stored on Azure.</param>
+        /// <param name="target">Stream: stream to contain the blob data.</param>
         /// <param name="cancellationToken">A System.Threading.CancellationToken to observe while waiting for a task to complete.</param>
         /// <returns>A System.Threading.Tasks.Task object that represents the asynchronous operation.</returns>
-        Task<Stream> GetAsync(string blobName, CancellationToken cancellationToken = default);
+        Task<bool> GetAsync(string blobId, 
+                            Stream target,
+                            CancellationToken cancellationToken = default);
 
         /// <summary>
         /// Retrieves an item from Azure Blob storage.
         /// </summary>
         /// <remarks>The item must have been json-serialized otherwise it cannot be deserialized properly.</remarks>
         /// <typeparam name="T">Type of item.</typeparam>
-        /// <param name="blobName">string: The identifier/name of this content to be stored on Azure.</param>
+        /// <param name="blobId">string: The identifier/name of this content to be stored on Azure.</param>
         /// <param name="cancellationToken">A System.Threading.CancellationToken to observe while waiting for a task to complete.</param>
         /// <returns>A System.Threading.Tasks.Task object that represents the asynchronous operation.</returns>
-        Task<T> GetAsync<T>(string blobName, CancellationToken cancellationToken = default);
+        Task<T> GetAsync<T>(string blobId, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Retrieves an item from Azure Blob storage.
+        /// </summary>
+        /// <remarks>The item must have been json-serialized otherwise it cannot be deserialized properly.</remarks>
+        /// <typeparam name="T">Type of item.</typeparam>
+        /// <param name="blobId">string: The identifier/name of this content to be stored on Azure.</param>
+        /// <param name="existingPropertiesOrMetaData">IList&lt;string&gt;: collection of keys to extract from the blob's properties and/or custom meta data.</param>
+        /// <param name="cancellationToken">A System.Threading.CancellationToken to observe while waiting for a task to complete.</param>
+        /// <returns>A System.Threading.Tasks.Task object that represents the asynchronous operation.</returns>
+        Task<BlobData<T>> GetAsync<T>(string blobId, 
+                                      IList<string> existingPropertiesOrMetaData = default, 

```

Comment:
Agreed with adding the overload vs bloating existing, but.. do you think we need the flexibility to specify which properties to return? 

Few reasons i'm thinking:
- They won't know which properties they need? (unless we use an enum?)
- There's no performance impact of setting some/all properties right? So there's no harm to set all the fields. If they don't need something, they can ignore.

So i'm thinking we just change this overload to remove this param.. or maybe `bool fetchProperties = false`?